v1/src/main/java/com/google/cloud/teleport/templates/common/SpannerConverters.java

```diff
@@ -442,6 +527,128 @@ private static List<String> parseResultSet(
+  public static class StructJSONPrinter implements StructPrinter {
+    private StructValidator structValidator = null;
+
+    public StructJSONPrinter(StructValidator structValidator) {
+      this.structValidator = structValidator;
+    }
+
+    public StructJSONPrinter() {}
+
+    /**
+     * To get string in json format from Struct.
+     *
+     * @param struct Spanner Struct.
+     * @return Spanner Struct encoded as a JSON String.
+     */
+    @Override
+    public String print(Struct struct) {
+      if (structValidator != null) {
+        structValidator.validate(struct);
+      }
+
+      StringWriter stringWriter = new StringWriter();
+      try {
+        JsonWriter jsonWriter = new JsonWriter(stringWriter);
+        LinkedHashMap<String, BiFunction<Struct, String, String>> parsers = Maps.newLinkedHashMap();
+        parsers.putAll(mapColumnParsers(struct.getType().getStructFields()));
+        parseResultSet(jsonWriter, struct, parsers);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+      return stringWriter.toString();
+    }
+
+    /**
+     * To parse Struct using different parsers for different type and process it using JSONWriter.
+     *
+     * @param jsonWriter JSON Writer
+     * @param struct Data from Spanner
+     * @param parsers Map from Column Name to their parser
+     * @throws IOException
+     */
+    private static void parseResultSet(
+        JsonWriter jsonWriter,
+        Struct struct,
+        LinkedHashMap<String, BiFunction<Struct, String, String>> parsers)
+        throws IOException {
+      // Start the JSON object
+      jsonWriter.beginObject();
+
+      for (String columnName : parsers.keySet()) {
+        LOG.info("Column Type: {}", struct.getColumnType(columnName).getCode());
+        String columnValue = parsers.get(columnName).apply(struct, columnName);
+
+        // If null value then ignoring it and not exporting to JSON file
+        if (columnValue == null) {
+          continue;
+        }
+
+        if (Arrays.asList(Code.JSON, Code.PG_JSONB)
+            .contains(struct.getColumnType(columnName).getCode())) {
+          // If the column is of type JSON or PG_JSON
+          jsonWriter.name(columnName).jsonValue(parsers.get(columnName).apply(struct, columnName));
+        } else if (struct.getColumnType(columnName).getCode() == Code.ARRAY) {
+          // If the column is of type ARRAY
+          List<? extends Object> values = parseArrayValueAsObjectList(struct, columnName);
+          jsonWriter.name(columnName);
+          jsonWriter.beginArray();
+          for (Object value : values) {
+            jsonWriter.value(value != null ? value.toString() : null);
+          }
+          jsonWriter.endArray();
+        } else if (parsers.containsKey(columnName)) {
+          jsonWriter.name(columnName).value(columnValue);
+        } else {
+          throw new RuntimeException("No parser for column: " + columnName);
```

Comment:
If this else block is a valid case, then it would have resulted in an error at L#582 itself right?

Reply:
Correct, it's a redundant check.